### PR TITLE
Remove unused local variables

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

This PR removes useless assignations. There are also somes fixes in tests.

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Tests/FormTest.php [hunk 1]
```diff
@@ -81,8 +89,10 @@ public function testConstructorThrowsExceptionIfNoRelatedForm()
 
         $nodes = $dom->getElementsByTagName('input');
 
-        $form = new Form($nodes->item(0), 'http://example.com');
-        $form = new Form($nodes->item(1), 'http://example.com');
```
Comment (fancyweb) [inline src/Symfony/Component/DomCrawler/Tests/FormTest.php]: The second one is never tested since the first one throws an exception.

**Review by yceruto — APPROVED** [review]

File: src/Symfony/Component/HttpKernel/HttpCache/ResponseCacheStrategy.php [hunk 2]
```diff
@@ -130,7 +130,6 @@ public function update(Response $response)
         $response->headers->set('Cache-Control', implode(', ', array_keys($flags)));
 
         $maxAge = null;
-        $sMaxage = null;
```
Comment (fancyweb) [inline src/Symfony/Component/HttpKernel/HttpCache/ResponseCacheStrategy.php]: `$sMaxage` is assigned line 140 and only used in the `if`.

File: src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php [hunk 3]
```diff
@@ -49,7 +49,6 @@ public function collect(Request $request, Response $response, \Exception $except
             }
         }
 
-        $content = null;
```
Comment (fancyweb) [inline src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php]: `$content` is directly assigned in the try and in the catch and there is not check on it.

File: src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php [hunk 4]
```diff
@@ -59,7 +58,6 @@ public function collect(Request $request, Response $response, \Exception $except
 
         $sessionMetadata = [];
         $sessionAttributes = [];
-        $session = null;
```
Comment (fancyweb) [inline src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php]: `$session` is assigned line 63 and only used in the `if`.

File: src/Symfony/Component/ExpressionLanguage/Tests/ExpressionLanguageTest.php [hunk 5]
```diff
@@ -196,7 +196,7 @@ public function testCachingWithDifferentNamesOrder()
         $cacheMock = $this->getMockBuilder('Psr\Cache\CacheItemPoolInterface')->getMock();
         $cacheItemMock = $this->getMockBuilder('Psr\Cache\CacheItemInterface')->getMock();
         $expressionLanguage = new ExpressionLanguage($cacheMock);
-        $savedParsedExpressions = [];
+        $savedParsedExpression = null;
```
Comment (fancyweb) [inline src/Symfony/Component/ExpressionLanguage/Tests/ExpressionLanguageTest.php]: This one is a fix on the test.

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 6]
```diff
@@ -219,7 +219,7 @@ public function dump(array $options = [])
                 foreach ($ids as $id) {
                     $c .= '    '.$this->doExport($id)." => true,\n";
                 }
-                $files['removed-ids.php'] = $c .= "];\n";
```
Comment (fancyweb) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: The resulting `$c` is never reused so the assignation is useless.

File: src/Symfony/Component/Debug/DebugClassLoader.php [hunk 7]
```diff
@@ -404,7 +404,7 @@ private function darwinRealpath($real)
         }
 
         if (isset($dirFiles[$file])) {
-            return $real .= $dirFiles[$file];
```
Comment (fancyweb) [inline src/Symfony/Component/Debug/DebugClassLoader.php]: Returning an assignation is useless, `$real` is not passed by ref.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 8]
```diff
@@ -1043,8 +1043,6 @@ private function registerAssetsConfiguration(array $config, ContainerBuilder $co
         $container->getDefinition('assets.url_package')->setPrivate(true);
         $container->getDefinition('assets.static_version_strategy')->setPrivate(true);
 
-        $defaultVersion = null;
```
Comment (fancyweb) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: `$defaultVersion` is always assigned in the if else block.

File: src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php [hunk 9]
```diff
@@ -425,7 +425,6 @@ private function renderEventListenerTable(EventDispatcherInterface $eventDispatc
         $tableHeaders = ['Order', 'Callable', 'Priority'];
         $tableRows = [];
 
-        $order = 1;
```
Comment (fancyweb) [inline src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/TextDescriptor.php]: `$order` is directly reassigned by the foreach.

File: src/Symfony/Bridge/Twig/Node/SearchAndRenderBlockNode.php [hunk 10]
```diff
@@ -28,7 +28,6 @@ public function compile(Compiler $compiler)
 
         preg_match('/_([^_]+)$/', $this->getAttribute('name'), $matches);
 
-        $label = null;
```
Comment (fancyweb) [inline src/Symfony/Bridge/Twig/Node/SearchAndRenderBlockNode.php]: `$label` is assigned line 42 and only used in this code path.

**Review by yceruto — APPROVED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @fancyweb.

**Pull request merged**